CreeDictionary/API/templatetags/inflection_extras.py

```diff
@@ -77,3 +79,36 @@ def presentational_pos(wordform: Union[Wordform, dict]) -> str:
         f"can not determine presentational pos for {wordform_dict}, id={wordform_dict['id']}"
     )
     return ""
+
+
+# XXX: mypy can't infer this type?
+PER_REQUEST_ID_COUNTER = WeakKeyDictionary()  # type: WeakKeyDictionary
+
+
+@register.filter
+def unique_id(context: Any) -> str:
```

Comment:
https://docs.djangoproject.com/en/3.0/howto/custom-template-tags/

So I think "filters"( `{{ ]}`) takes arbitrary value, and "tags" (`{% %}`) are the ones that are meant to take in context? You may want to use django's `@register.simple_tag(takes_context=True)` here

Reply:
I debated A LOT about this. You're right; context makes more sense, but the fact that I needed to assign and recall the ID twice makes creating a template tag... annoying... I'll give it some more thought :/